src/pages/Home/Home.js

```diff
@@ -1,24 +1,102 @@
-import React from 'react';
 import { useHistory } from 'react-router-dom';
+import { useState } from 'react';
 
-function Home({ userToken, saveToken }) {
+function Home({ createList, joinList }) {
   let history = useHistory();
 
-  function handleClick(e) {
-    saveToken(e);
+  const [shareToken, setShareToken] = useState('');
+  const [joinListError, setJoinListError] = useState('');
 
-    if (userToken) {
-      history.push('/list');
-    }
+  const handleTokenChange = (event) => {
+    setJoinListError('');
+    setShareToken(event.target.value);
+  };
+
+  function handleCreateList() {
+    createList()
+      .then((success) => {
+        history.push('/list');
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  function handleJoinList(event) {
+    event.preventDefault();
+    setJoinListError(''); // remove message to ensure repeated error is read again by screen reader
+    joinList(shareToken)
```

Comment:
Where does  `shareToken` come from that is passed into `joinList` here? It is the user-provided token? For my understanding, where are you getting the token from in this function?

Reply:
Are we essentially passing in the user's token into `joinList`? I'm not sure I intuitively understood the variable name `shareToken` at first. Can we just call it userToken? The more I look at it, the more it makes sense to me as the token used in order to share...but just my initial thoughts!

Reply:
`shareToken` is the value of the form field labelled "Share Token" in the wireframe, and with the name and id of "shareToken"—basically we used the exact same pattern of a controlled form field component as in the "Add Item" form, but since there will only ever be one input we named the input's linked state variable consistent with the input. Given that, I think shareToken (or maybe shareTokenValue) makes sense, but if you still feel like userToken is clearer I think we should change it in the id/name/classes as well to be consistent.

Reply:
Yes, `shareToken` is the token the user enters in the bottom form field on the Home screen to join a list. And yes, that user provided token is passed to `joinList`. The "share token" name comes from the [wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png) provided for this story, as Ander said. Given that, I think I would err on the side of keeping it as `shareToken`, because I assume future project stories might refer to it as that?

Reply:
Looking at Nick's comment and looking at the next few stories that we have coming up, I think it is ok to keep it as shareToken or shareTokenValue, since now we have removed userToken an replaced with createList and joinList.

Reply:
Just stuck in some comments to further clarify the state variables we are using, since I added another one for a11y updates. I think it should be clearer now.